File: java/leetcode/src/main/java/com/leetcode/graph/Keys_and_Rooms_841.java
package com.leetcode.graph;

import java.util.List;

public class Keys_and_Rooms_841 {
    public boolean canVisitAllRooms(List<List<Integer>> rooms) {
        int n = rooms.size();
        boolean[] visited = new boolean[n];
        dfs(rooms, 0, visited);
        for (boolean v : visited) {
            if(!v) return false;
        }
        return true;
    }

    void dfs(List<List<Integer>> rooms, int i, boolean[] visited) {
        if (visited[i]) return;
        visited[i] = true;
        List<Integer> keys = rooms.get(i);

        for(Integer key : keys) {
            dfs(rooms, key, visited);
        }

    }
}

File: js/algrithm/LeetCode 75/Graphs - DFS/Keys and Rooms_841.js
var canVisitAllRooms = function(rooms) {
    const n = rooms.length
    const visited = Array(n).fill(false)
    const dfs = function(i, visited) {
     visited[i] = true
     for (let next of rooms[i]) {
         if (!visited[next]) dfs(next, visited)
     }
    }
 
    dfs(0, visited)
    return visited.filter(v => !v).length === 0
 };